WordPress/src/main/java/org/wordpress/android/ui/prefs/notifications/NotificationsSettingsDialogFragment.kt

```diff
@@ -0,0 +1,399 @@
+package org.wordpress.android.ui.prefs.notifications
+
+import android.annotation.SuppressLint
+import android.app.Dialog
+import android.content.DialogInterface
+import android.os.Bundle
+import android.text.TextUtils
+import android.view.View
+import android.view.ViewGroup
+import android.widget.CompoundButton
+import android.widget.LinearLayout
+import android.widget.ScrollView
+import android.widget.TextView
+import androidx.appcompat.app.ActionBar
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.widget.SwitchCompat
+import androidx.core.content.ContextCompat
+import androidx.fragment.app.DialogFragment
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import org.json.JSONException
+import org.json.JSONObject
+import org.wordpress.android.R
+import org.wordpress.android.databinding.NotificationsSettingsSwitchBinding
+import org.wordpress.android.models.NotificationsSettings
+import org.wordpress.android.ui.prefs.AppPrefs
+import org.wordpress.android.util.AppLog
+import org.wordpress.android.util.JSONUtils
+import org.wordpress.android.util.extensions.getDrawableFromAttribute
+
+class NotificationsSettingsDialogFragment(
+    private val channel: NotificationsSettings.Channel,
+    private val type: NotificationsSettings.Type,
+    private val blogId: Long = 0,
+    private val settings: NotificationsSettings,
+    private val onNotificationsSettingsChangedListener:
+        NotificationsSettingsDialogPreference.OnNotificationsSettingsChangedListener,
+    private val bloggingRemindersProvider: NotificationsSettingsDialogPreference.BloggingRemindersProvider? = null,
+    private val title: String
+): DialogFragment(), PrefMainSwitchToolbarView.MainSwitchToolbarListener, DialogInterface.OnClickListener {
+    companion object {
+        const val TAG = "Notifications_Settings_Dialog_Fragment"
+        private const val SETTING_VALUE_ACHIEVEMENT = "achievement"
+    }
+
+    private val mUpdatedJson = JSONObject()
+    private var mTitleViewWithMainSwitch: ViewGroup? = null
+
+    // view to display when main switch is on
+    private var mDisabledView: View? = null
+
+    // view to display when main switch is off
+    private var mOptionsView: LinearLayout? = null
+
+    private var mMainSwitchToolbarView: PrefMainSwitchToolbarView? = null
+    private val mShouldDisplayMainSwitch: Boolean = settings.shouldDisplayMainSwitch(channel, type)
+
+    private var mSettingsArray = arrayOfNulls<String>(0)
+    private var mSettingsValues: Array<String?>? = arrayOfNulls<String?>(0)
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val context = requireContext()
+        @SuppressLint("InflateParams")
```

Comment:
We could avoid the use the suppress by using the `onCreateView` method. [This is an example](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/ui/activitylog/list/filter/ActivityLogTypeFilterFragment.kt#L66) `DialogFragment` that does this.